### DISCUSSION TITLE
Missing file when using adapter/static as pwa with dynamic routes

### DISCUSSION THREAD

**Discussion by m1212e** [sveltejs/kit#10024] [Q&A]
Hi,

I'm using the static adapter to generate static files for my app. I have a few dynamic routes. E.g. I have `files/owned` and `files/shared` which display a list of files. Since the code behind them is pretty similar and it is likely that there will be more filters in the future, I put those behind a dynamic route `files/[filter]`. I export an entries property to make Sveltekit aware of them so they get prerendered:

```
export const entries = (() => {
    return availableCategories().map(category => {
        return {filter: category}
    })
}) satisfies EntryGenerator;
```

This works great.
Now I want to make a dynamic route with an unknown amout of variations. E.g. `filebrowser/[fileId]`. To make the app build, I disabled the prerender option for this route: `export const prerender = false;`. The app builds and everything works fine if I navigate from any page in the app to my file browser, because the client side router handles everything. But when I navigate to my file browser as initial page visit (no client side router to handle and intercept the navigation) then the app breaks. Initially I thought this was due to some misconfiguration of the fallback page. I've set `fallback: '200.html'` in my adapter configuration and I serve the `200.html` on every not found file. Now when I visit the page with my filebrowser as an entrypoint, I get a white screen. Looking into the network tab, I see only successful requests. My console says `Uncaught (in promise) SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data`. I think this happens, because the browser requests `filebrowser/1234/__data.json` which does not exist and gets answered with my `200.html`.

How do I fix this and what even is the `__data.json` I'm missing?

Thanks for your time & help!

**Comment by teemingc** [comment]
Can you provide an example repository?

**Reply by m1212e** [reply]
Sure: https://github.com/m1212e/svelte-kit-bug-demo-

To replicate, please install and build. Then run the provided server.cjs file to spin up a simple express server serving the build directory.
To trigger the error, navigate to `/search/1234`. This should log the parse error to the console.

I think it's caused by https://github.com/ivanhofer/typesafe-i18n but I can't figure out what is misconfigured. I followed their SvelteKit Tutorial. I don't really understand what the missing data.json does.

**Reply by teemingc** [reply]
I can't seem to replicate this. The page loads correctly for me.
